Created service for integration_connectors

SamuelYip29 opened the pull request:
Finished the service integration_connectors, will complete the documentation next week.

SamuelYip29 committed:
Add integration_connectors policy

SamuelYip29 committed:
Added documentation

visal-dam commented on policies/gcp/integration_connectors/google_integration_connectors_connection/secret_manager_secret/policy.rego:
Is the attribute path right here?

visal-dam replied:
Attribute path is still incorrect; "google_secret_manager_secret_iam_member", 0,"member" is what I think you mean here.

visal-dam commented on policies/gcp/integration_connectors/google_integration_connectors_connection/user_password/policy.rego:
I think you're also missing a ".secret-version.name" here.

visal-dam replied:
Line 13 is misspelt.

visal-dam commented on policies/gcp/integration_connectors/google_integration_connectors_endpoint/service_attachment/policy.rego:
Good, though can be expanded to pattern whitelist specific projects, such as "values": ["projects/*/"], [["connectors-example1", "connectors-example2", etc.]]

visal-dam replied:
You're almost correct with the patterns, but as you have two \*, you need two pattern arrays afterwards to map them according to their position. Your current policy's patterns only affect that first project/\*/regions/.... and not the last one. I think this pattern is only for the last \*, i.e., project/my-project/regions/us-cebtral1/serviceAttachments/"connectors-example1" OR "connectors-example2" OR "connectors-example3". So you can replace the first * with "my-project" or some other placeholder.

visal-dam reviewed (changes requested):
Excellent policy structuring. Really minor suggestions. Also, your PR includes a "cloud_vmware_engine" which should not be there - please remove it. Everything else seems good.

SamuelYip29 committed:
feedback addressed

SamuelYip29 commented:
I've addressed the feedback and deleted the cloud_vmare_engine from my vs code, but I seems to still see it as the issue "This branch has conflicts that must be resolved" appears above, this is normal? Thx.

visal-dam reviewed (changes requested):
Minor changes. Make sure you test your policies for correctness before submitting them as a PR.

SamuelYip29 committed:
Feedback addressed

SamuelYip29 committed:
Merge branch 'dev' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/service/integration_connectors

SamuelYip29 committed:
Revert "Merge branch 'dev' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/service/integration_connectors"

This reverts commit e818ac78016477a1ab67d557fa7d8bec9fb04a38, reversing
changes made to 59c1dd488204f9c154db36ff99d2acba9efc5f62.

visal-dam commented:
Closed. Replaced by new branch without merge conflicts.
